python/cudf/cudf/core/indexing_utils.py

```diff
@@ -0,0 +1,225 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+from __future__ import annotations
+
+import enum
+import itertools
+from typing import TYPE_CHECKING, Any, List, Tuple, Union
+
+import numpy as np
+from typing_extensions import TypeAlias
+
+import cudf
+import cudf._lib as libcudf
+from cudf.api.types import (
+    _is_scalar_or_zero_d_array,
+    is_bool_dtype,
+    is_integer,
+    is_integer_dtype,
+)
+
+if TYPE_CHECKING:
+    from cudf.core.column import ColumnBase
+
+
+class IndexTag(enum.IntEnum):
```

Comment:
How about:

```suggestion
class IndexerTag(enum.IntEnum):
```

`Index` is a bit overloaded in Pandas, and I believe indexer is the more appropriate term here anyway.

Reply:
Ditto below for `IndexSpec`.

Reply:
Went for `Indexing` but can be persuaded to use `Indexer` as well.